apache/cloudstack
# Extend import/export unmanaged instances to KVM

This PR adds supports to cloudstack for managing and unmanaging VM's in KVM Hypervisor. 

This PR addresses  #7127

### Description

The support is extended through API's and also the UI. The Apache Cloudstack  KVM plugin has been extended with new command wrappers for fetching unmanaged VM's xml attribute. The code paths in management server have been cleaned up and refined for managing KVM specific importing commands.
API's involved are :

- unmanageVirtualmachine
- listunmanageInstances
- importVirtualMachine

### Testing steps:

- Build pull request and deploy a new apache cloudstack environment with KVM hosts(used monkeybox for local development and testing).
- From the cloudstack UI, create a new virtual machine.
- Go to tools option and choose import-export instances.
- In the managed instance section, see if the newly created virtual machine exist.
- From the managed instance section, choose the virtual machine and unmanaged it.
- Once the VM is unmanaged you can see it disappear from manage instance section and it will appear in unmanage instance section.
- To manage an unmanage instance, choose an instance from unmanage instance section.
- On click of unmanage  instance a popup will appear, where you need to enter vm name and choose vm template, service offering, network and then click import instance button.
- You can see the unmanage instance is managed now and you can see it populate in the mananged instance section under import-export instances section.

















### Types of changes


### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale


#### Bug Severity



### Screenshots (if appropriate):


### How Has This Been Tested?
